modules/siddhi-extensions/time/src/main/java/org/wso2/siddhi/extension/time/ExtractDayOfWeekFunctionExtension.java

```diff
@@ -0,0 +1,170 @@
+                    " but found " + attributeExpressionExecutors[0].getReturnType().toString());
+        }
+        //User can omit sending the dateFormat thus using a default CEP Time format
+        if (attributeExpressionExecutors.length > 0) {
+            if (attributeExpressionExecutors.length > 1 && attributeExpressionExecutors[1].getReturnType() != Attribute.Type.STRING) {
+                throw new ExecutionPlanValidationException("Invalid parameter type found for the second argument of " +
+                        "time:dayOfWeek(dateValue,dateFormat) function, " + "required " + Attribute.Type.STRING +
+                        " but found " + attributeExpressionExecutors[1].getReturnType().toString());
+            }
+        }
+
+    }
+
+    // this method will be executed for two or more parameters
+    @Override
+    protected Object execute(Object[] data) {
+        String userFormat;
+        if (data[0] == null) {
+            throw new ExecutionPlanRuntimeException("Invalid input given to time:dayOfWeek(dateValue," +
+                    "dateFormat) function" + ". First " + "argument cannot be null");
+        }
+        if (data.length > 1) {
+            if (data[1] == null) {
+                throw new ExecutionPlanRuntimeException(
+                        "Invalid input given to time:dayOfWeek(dateValue,dateFormat) function" + ". Second " +
+                                "argument cannot be null");
+            } else {
+                userFormat = (String) data[1];
+            }
+        } else {
+            userFormat = TimeExtensionConstants.EXTENSION_TIME_DEFAULT_DATE_FORMAT;
+        }
+        String source = null;
+        FastDateFormat userSpecificFormat;
+        Date userSpecifiedDate;
+        try {
+            source = (String) data[0];
+            userSpecificFormat = FastDateFormat.getInstance(userFormat);
+            userSpecifiedDate = userSpecificFormat.parse(source);
+            return getDayOfWeek(userSpecifiedDate);
+        } catch (ParseException e) {
+            String errorMsg = "Provided format " + userFormat + " does not match with the timestamp " + source + e
+                    .getMessage();
+            throw new ExecutionPlanRuntimeException(errorMsg, e);
+        } catch (ClassCastException e) {
+            String errorMsg = "Provided Data type cannot be cast to desired format. " + e.getMessage();
+            throw new ExecutionPlanRuntimeException(errorMsg, e);
+        }
+    }
+
+
+    // this method will be executed for a single parameter
+    @Override
+    protected Object execute(Object data) {
+        String userFormat;
+        if (data == null) {
+            throw new ExecutionPlanRuntimeException("Invalid input given to time:dayOfWeek(dateValue," +
+                    "dateFormat) function" + ". First " + "argument cannot be null");
+        }
+        userFormat = (String) data;
```

Comment:
@minudika 
In this method, it seems you are assigning the same value for both dateFormat and dateValue. Since this method gets executed only for a single parameter, I think the dateFormat should be fixed and only the dateValue should be considered.